src/systems/filecoin_blockchain/struct/chain/chain.go

```diff
@@ -18,8 +18,17 @@ func (chain *Chain_I) TipsetAtEpoch(epoch abi.ChainEpoch) Tipset {
 
 // Draws randomness from the tipset at or immediately prior to `epoch`.
 func (chain *Chain_I) RandomnessAtEpoch(epoch abi.ChainEpoch) abi.RandomnessSeed {
-	ts := chain.TipsetAtEpoch(epoch)
-	return ts.MinTicket().DrawRandomness(epoch)
+	if epoch < genesis {
+		genesisTS := chain.TipsetAtEpoch(genesis)
+		genesisTix := genesisTS.MinTicket().DrawRandomness(genesis)
+		buffer := []byte{}
+		buffer = append(buffer, genesisTix...)
+		buffer = append(buffer, BigEndianBytesFromInt(int64(epoch))...)
+		return blake2b.Sum256(buffer)
+	} else {
+		ts := chain.TipsetAtEpoch(epoch)
```

Comment:
Please leave a big comment here describing that the epoch is intentionally not mixed in because some uses of randomness don't need it. It is at the caller's peril to mix in the chain epoch in order to reliably get different randomness for each epoch (given null blocks). But note that in cases of pre-genesis sampling this will be redundant. 

I think this should either always or never mix in the epoch number.

Reply:
Changed to always per our convo. Thank you